### PR TITLE
Upstream 23.03

### PR DESCRIPTION
**Pull request by RassK**
### Ducktype: Adds module isolation by target assembly [(#1291)](https://github.com/DataDog/dd-trace-dotnet/pull/1291)

* Adds module isolation per target assembly
* formatting
* Changes based in the review.
* Changes.
* Add DuplicateTypeProxy smoke test.
* Adds missing DuplicateTypeProxy in linux build scripts.
* fix linux target frameworks.
* Removing TargetFrameworks to avoid CI issues.

### Update changelog for 1.25.0 [(#1298)](https://github.com/DataDog/dd-trace-dotnet/pull/1298)

* Update changelog for 1.25.0

### Add additional ASP.NET Core tests + fix response code bug [(#1269)](https://github.com/DataDog/dd-trace-dotnet/pull/1269)

* Move SerializableDictionary to TestHelpers library
* Add additional AspNetCoreDiagnosticObserver tests

Adds tests for tags set (none) and resource names, in preparation for updating these later behind a feature flag

Required so we can reference them directly from test apps

* Add testing of response and span status code in Diagnostic Source
* Add a Razor Pages sample application for diagnostic observer tests
* Add tests for AspNetCore RazorPages
* Add tests for ApiController endpoints
* Add testing of unhandled exception handling

Tests several different scenarios:
- No error handling middleware, HttpServer handles the exception
- Standard developer exception page
- Error handling middleware with pipeline reexecution
- Custom Handler, which only catches a subset of errors

Used multiple Mapped pipelines to make testing the different paths easier. Hardest thing was hitting the edge case path in Kestrel for when the request is malformed, as that was made internal in 3.0, but is still worth testing I think

* Add tests for exceptions in non-mvc pipelines
* Fix ASP.NET core DiagnosticObserver recording wrong status code for unhandled exceptions

If there's no error handling middleware, the HttpServer (e.g. Kestrel) handles the exception, and updates the response status code. Although the unhandled exception callback in diagnostic observer is invoked, the status code still has a 200 status code at this point, as Kestrel doesn't update it to 500/4X error until later

To work around this, we assume a status code of 500 (or use the value in the internal BadHttpRequestException), and _don't_ update the status code on request end.

* Skip re-execute tests for now

It's not possible to fix these cases without more extensive work. These can be fixed at the same time as we fix the overall resource names.

* Move tests to integration library
* Fix build configurations
* Strong name the RazorPages project that is directly referenced from test project
* Set the application parts for TestServer tests

Ensures we don't accidentally pick up the Razor Pages app

* Fix Datadog.Trace.IntegrationTests output type
* Update StatusCode API based on PR comments

### Fix Container Tagging in Fargate 1.4 [(#1286)](https://github.com/DataDog/dd-trace-dotnet/pull/1286)

* Refactor ContainerIDRegex to make it easier to understand
* Format of cgroups file was changed in Fargate 1.4.

 We now extract the 'task id' from cgroups file, and report as 'Datadog-Container-ID' tag by tracers.

### Minor build improvements [(#1295)](https://github.com/DataDog/dd-trace-dotnet/pull/1295)

* Add some more CheckEolTargetFramework
* Delete unused sample project

### CODE REVIEW

File: integrations.json [hunk 1]
```diff
@@ -3555,7 +3555,7 @@
           "maximum_patch": 65535
         },
         "wrapper": {
-          "assembly": "OpenTelemetry.AutoInstrumentation.ClrProfiler.Managed, Version=1.24.0.0, Culture=neutral, PublicKeyToken=def86d061d0d2eeb",
+          "assembly": "OpenTelemetry.AutoInstrumentation.ClrProfiler.Managed, Version=1.24.0.0, Culture=neutral, PublicKeyToken=34b8972644a12429",
```
Comment (pjanotti) [inline integrations.json:3558]: 👍

File: test/test-applications/regression/DuplicateTypeProxy/Properties/launchSettings.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    "DuplicateTypeProxy": {
+      "commandName": "Project",
+      "environmentVariables": {
+        "COR_ENABLE_PROFILING": "1",
+        "COR_PROFILER": "{918728DD-259F-4A6A-AC2B-B85E1B658318}",
+        "COR_PROFILER_PATH": "$(ProjectDir)$(OutputPath)profiler-lib\\Datadog.Trace.ClrProfiler.Native.dll",
+
+        "CORECLR_ENABLE_PROFILING": "1",
+        "CORECLR_PROFILER": "{918728DD-259F-4A6A-AC2B-B85E1B658318}",
+        "CORECLR_PROFILER_PATH": "$(ProjectDir)$(OutputPath)profiler-lib\\Datadog.Trace.ClrProfiler.Native.dll",
```
Comment (pjanotti) [inline test/test-applications/regression/DuplicateTypeProxy/Properties/launchSettings.json:12]: These were renamed. This is something that we could automate via a tool, but not for this PR.

Is it covered in the instructions? If not could you please add it there.

**Review by pjanotti — COMMENTED** [review]
@RassK just one minor correction on a launchSettings.json

File: test/test-applications/regression/DuplicateTypeProxy/Properties/launchSettings.json (hunk 2 above)
Comment (RassK) [reply test/test-applications/regression/DuplicateTypeProxy/Properties/launchSettings.json]: Fixed

**Review by pjanotti — APPROVED** [review]
LGTM
